README.md

```diff
@@ -0,0 +1,163 @@
+  whenever a device might need some initialisation sequence, such as server
+  start up or error recovery. It is up to the script to detect whether it needs
+  to actually do anything or not. It can be left empty, `""`, if no such script
+  is needed.
+
+#### Trigger configuration
+
+A trigger can be configured using
+
+- `interval` , a 32bit unsigned integer, which specifies the time between ticks
+  in milliseconds.
+
+#### Walkthrough of a full example
+
+The code snipptes below are taken from the XML file in the `sample_configs`
+folder in this repository.
+
+First, we start with the usual XML declaration and root tag.:
+
+```xml
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<configuration>
+```
+
+We then declare that we want to connect two devices, `TMCB1` and `TMCB2` and
+have one periodic timer that is called `msTimer`.
+
+```xml
+  <variable name="devices" type="string">
+    <value i="0"  v="TMCB1"/>
+    <value i="1"  v="TMCB2"/>
+  </variable>
+
+  <variable name="periodicTimers" type="string">
+    <value i="0"  v="msTimer"/>
+  </variable>
+```
+
+The timer should run every 1000 milliseconds
+
+```xml
+  <module name="msTimer">
+    <variable name="period" type="uint32" value="1000" />
+  </module>
+```
+
+For the device `TMCB1`, we want it to be read out once a second using the
+`msTimer` trigger, but are only interested in the `BSP` part of the device. Upon
+start or recovery, we need to do additional initialisation of the device using
+a script called `writeWordStatus`, which is located next to the server binary.
+
+```xml
+ <module name="TMCB1">
+    <variable name="triggerPath" type="string" value="/msTimer/tick" />
+    <variable name="pathInDevice" type="string" value="/BSP" />
+    <variable name="initScript" type="string" value="./writeWordStatus" />
+ </module>
+```
+
+The device TMCB2, on the other hand, shall only be read on manual request, for
+example if someone might click on a button in the user interface of the control
+system. Therefore we set the trigger path to `/manual/trigger`. We want to
+expose the whole device and do not need any additional initialisation, so we
+configure those empty.
+
+```xml
+  <module name="TMCB2">
+    <variable name="triggerPath" type="string" value="/manual/trigger" />
+    <variable name="pathInDevice" type="string" value="" />
+    <variable name="initScript" type="string" value="" />
+ </module>
+```
+
+And then we just close off the file.
+
+```xml
+</configuration>
+```
+
+### Configuring the control system
+
+The configuration file for the control system depends on the chosed control
+system adapter. If no file is provided, it will default to expose everything to
+the control system.
+
+#### DOOCS
+
+The file for DOOCS has to be named
+`generic_chimeratk_server-DoocsVariableConfig.xml`. It might also be advisable,
+though not necessary, to provide a DOOCS server configuration file containing
```

Comment:
It is necessary to have a doocs server config to specify the RPC number. The DoocsVariableConfig.xml however is optional.

Reply:
Reworded the DOOCS section